reference/services/buildbot.rst

```diff
@@ -0,0 +1,158 @@
+Buildbot
+========
+
+Short description
+-----------------
+
+`Buildbot <https://github.com/buildbot/buildbot>`__ is a Python-based
+continuous integration testing framework that we use to run tests,
+manage stable branches for Launchpad. We use `Buildbot
+0.8.12 <https://docs.buildbot.net/0.8.12/>`__ in production currently.
+You can access our hosted instance at http://lpbuildbot.canonical.com
+
+Detailed description
+--------------------
+
+There are the following main components in the Launchpad Buildbot
+setup. 
+
+1. `lpbuildbot <https://code.launchpad.net/~canonical-launchpad-branches/lpbuildbot/>`__:
+   This contains the \`master.cfg\` file and the related Python code
+   which powers the
+   `buildmaster <https://docs.buildbot.net/0.8.12/manual/introduction.html>`__
+   in our buildbot setup. There is a public bazaar branch and a private
+   production branch, which contains some secrets related to our
+   production deployment. We add changes to the public branch via merge
+   proposals and then merge them back to the private “production” branch
+   before deployment. In the ``master.cfg`` file, we have defined all
+   the necessary configurations (schedulers, workers, job steps) for
+   buildbot. 
+
+   1. The buildmaster has a custom ``buildbot-poll.py`` cron script
+      which is responsible for merging the commits in the
+      \`master`/\`db-devel\` branches, that have completed a successful
```

Comment:
I'd add a 'within the main Lauchpad repository' to make it clear that this is referring to the Launchpad repo branches